### ISSUE TITLE
reading into a const pointer member

### ISSUE THREAD

**Issue by ryan-tin** [stephenberry/glaze#1384]
I am not sure how to read to a member that is a `const` pointer. That is, the pointer is `const`, but the object itself is mutable.
Here's a minimal example:

```cpp
struct Address {
  std::string street;
}; // automatic reflection

class Person {
public:
  Person(Address *const p_add) : p_add(p_add) {};
  std::string name;
  Address *const p_add; // pointer is const, Address object is mutable
};

template <> struct glz::meta<Person> {
  using T = Person;
  static constexpr auto value = object(&T::name, &T::p_add);
};

int main() {
  std::string buffer = R"({"name":"Foo Bar","p_add":{"street":"Baz Yaz"}})";
  Address add{};
  Person p{&add};
  auto ec = glz::read<glz::opts{.format = glz::JSON, .error_on_const_read = true}>(p, buffer);
  if (ec) {
    std::cout << glz::format_error(ec, buffer) << std::endl;
  }
  assert(p.name == "Foo Bar");
  assert(p.p_add->street == "Baz Yaz");
}
```
the output when compiling and running
```
❯ ./main
1:27: attempt_const_read
   {"name":"Foo Bar","p_add":{"street":"Baz Yaz"}}
                             ^
Assertion failed: (p.p_add->street == "Baz Yaz"), function main, file main.cpp, line 32.
```
I don't believe that this is intended behavior because reading works if the `const` pointer is not a member:
```cpp
struct Address {
  std::string street;
};

int main() {
  std::string buffer = R"({"street":"1st street"})";

  Address add2{};
  Address *const p_add2 = &add2;
  auto ec =
      glz::read<glz::opts{.format = glz::JSON, .error_on_const_read = true}>(*p_add2, buffer);
  if (ec) {
    std::cout << glz::format_error(ec, buffer) << std::endl;
  }
  assert(p_add2->street == "1st street");
  std::cout << p_add2->street << std::endl;
}
```
I may be missing something, any advice appreciated!

**Comment by stephenberry** [comment]
Yeah, this looks like a bug. I'll try to fix it soon.

**Comment by stephenberry** [comment]
A fix (#1387) has been merged into `main`

**Comment by ryan-tin** [comment]
Thanks for the quick fix!
